consensus/tendermint/tendermint_test.go

```diff
@@ -0,0 +1,282 @@
+
+		expectedHeight, expectedRound := uint(0), uint(0)
+		expectedProposalMsg := Proposal[value, felt.Felt, felt.Felt]{
+			Height:     0,
+			Round:      0,
+			ValidRound: nil,
+			Value:      utils.Ptr(app.cur + 1),
+			Sender:     *nodeAddr,
+		}
+
+		proposalBroadcaster := broadcasters.ProposalBroadcaster.(*senderAndReceiver[Proposal[value, felt.Felt,
+			felt.Felt], value, felt.Felt, felt.Felt])
+
+		algo.Start()
+		algo.Stop()
+
+		proposal := <-proposalBroadcaster.mCh
+
+		assert.Equal(t, expectedProposalMsg, proposal)
+		assert.Equal(t, 1, len(algo.messages.proposals[expectedHeight][expectedRound][*nodeAddr]))
+		assert.Equal(t, expectedProposalMsg, algo.messages.proposals[expectedHeight][expectedRound][*nodeAddr][0])
+
+		assert.Equal(t, propose, algo.state.step)
+		assert.Equal(t, expectedHeight, algo.state.height)
+		assert.Equal(t, expectedRound, algo.state.round)
+	})
+
+	t.Run("node is not the proposer: schedule timeoutPropose", func(t *testing.T) {
+		listeners, broadcasters := testListenersAndBroadcasters()
+		app, chain, vals := newApp(), newChain(), newVals()
+
+		vals.addValidator(*val2)
+		vals.addValidator(*val3)
+		vals.addValidator(*val4)
+		vals.addValidator(*nodeAddr)
+
+		algo := New[value, felt.Felt, felt.Felt](*nodeAddr, app, chain, vals, listeners, broadcasters, tm, tm, tm)
+
+		algo.Start()
+		algo.Stop()
+
+		assert.Equal(t, 1, len(algo.scheduledTms))
+		scheduledTm := algo.scheduledTms[0]
+
+		assert.Equal(t, propose, scheduledTm.s)
+		assert.Equal(t, uint(0), scheduledTm.h)
+		assert.Equal(t, uint(0), scheduledTm.r)
+
+		assert.Equal(t, propose, algo.state.step)
+		assert.Equal(t, uint(0), algo.state.height)
+		assert.Equal(t, uint(0), algo.state.round)
+	})
+
+	t.Run("OnTimeoutPropose: round zero the node is not the proposer thus send a prevote nil", func(t *testing.T) {
+		listeners, broadcasters := testListenersAndBroadcasters()
+		app, chain, vals := newApp(), newChain(), newVals()
+		// The algo needs to run for a minimum amount of time but it cannot be long enough the state to change
+		// multiple times. This can happen if the timeouts are too small.
+		tm := func(r uint) time.Duration {
+			if r == 0 {
+				fmt.Println("R is ", r)
+				return time.Nanosecond
+			}
+			return time.Second
+		}
+
+		vals.addValidator(*val2)
+		vals.addValidator(*val3)
+		vals.addValidator(*val4)
+		vals.addValidator(*nodeAddr)
+
+		algo := New[value, felt.Felt, felt.Felt](*nodeAddr, app, chain, vals, listeners, broadcasters, tm, tm, tm)
+
+		expectedHeight, expectedRound := uint(0), uint(0)
+		expectedPrevoteMsg := Prevote[felt.Felt, felt.Felt]{
+			Vote: Vote[felt.Felt, felt.Felt]{
+				Height: 0,
+				Round:  0,
+				ID:     nil,
+				Sender: *nodeAddr,
+			},
+		}
+
+		prevoteBroadcaster := broadcasters.PrevoteBroadcaster.(*senderAndReceiver[Prevote[felt.Felt, felt.Felt], value,
+			felt.Felt, felt.Felt])
+
+		algo.Start()
+		time.Sleep(time.Millisecond)
+		algo.Stop()
```

Comment:
Removing these two lines seems to have no effect. Which is weird, I would have thought there were needed, and don't understand how algo.Stop() is triggered..